include/onnxruntime/core/platform/ort_mutex.h

```diff
@@ -108,7 +108,14 @@ std::cv_status OrtCondVar::wait_for(std::unique_lock<OrtMutex>& cond_mutex,
 namespace onnxruntime {
 
 class OrtMutex {
+#if defined(__clang__) && __cplusplus >= 202002L
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Wdeprecated-pragma"
```

Comment:
I think this can be fixed - see this patch: https://github.com/microsoft/onnxruntime/pull/22168/commits/e34c91c35bc9c274378e7aef88e67479ca12962c